catalog/transformer/catalog-transformer-overlay/src/main/java/ddf/catalog/transformer/OverlayMetacardTransformer.java

```diff
@@ -71,29 +83,106 @@ private BinaryContent overlay(Metacard metacard, Map<String, Serializable> argum
+        BufferedImage tile = createTileFromImageAndBoundary(image, boundary);
+        return createBinaryContent(tile);
     }
 
-    private void rotate(Builder imageBuilder, String wkt) throws CatalogTransformerException {
-        final double angle = GeometryUtils.getRotationAngle(wkt);
-        imageBuilder.rotate(angle);
+    private List<Vector> parseBoundary(String location) throws CatalogTransformerException {
+        final Geometry geometry = parseGeometry(location);
+        if (!canHandleGeometry(geometry)) {
+            throw new CatalogTransformerException("The metacard geometry is not a rectangle.");
+        }
+
+        final Coordinate[] coordinates = geometry.getCoordinates();
+
+        List<Vector> boundary = new ArrayList<>();
+        boundary.add(new BasicVector(new double[] {coordinates[0].x, coordinates[0].y}));
+        boundary.add(new BasicVector(new double[] {coordinates[1].x, coordinates[1].y}));
+        boundary.add(new BasicVector(new double[] {coordinates[2].x, coordinates[2].y}));
+        boundary.add(new BasicVector(new double[] {coordinates[3].x, coordinates[3].y}));
+        return boundary;
     }
 
-    private BinaryContent createBinaryContent(Builder imageBuilder)
-            throws CatalogTransformerException {
-        final ByteArrayOutputStream baos = new ByteArrayOutputStream();
+    private BufferedImage createTileFromImageAndBoundary(BufferedImage image, List<Vector> boundary) {
+        /*
+         * We transform the image by moving the corners and applying
+         * transparency so that it looks right when laid down as a north-up
+         * rectangular tile (the only thing Cesium currently supports).
+         */
+
+        // Scaling by latitude so our x and y axes have roughly equal units.
+        boundary.replaceAll(v -> scaleByLatitude(v));
 
+        // We are putting the image into a north-up rectangle, so we need
+        // to get the minimum rectangle surrounding the boundary.
+        List<Vector> boundingBox = calculateBoundingBox(boundary);
+        Vector origin = boundingBox.get(0).copy();
+        boundary.replaceAll(v -> v.subtract(origin));
+        boundingBox.replaceAll(v -> v.subtract(origin));
+
+        // The image may be stretched in weird ways, but we do our best to preserve
+        // the resolution by scaling by the width of the image when going from lon/lat
+        // to pixel space.
+        double scaleFactor = calculateScaleFactor(boundary, image.getWidth());
+        boundary.replaceAll(v -> v.multiply(scaleFactor));
+        boundingBox.replaceAll(v -> v.multiply(scaleFactor));
+
+        return createImage(image, boundary, (int) boundingBox.get(1).get(0), (int) -boundingBox.get(1).get(1));
+    }
+
+    private BinaryContent createBinaryContent(BufferedImage image)
+            throws CatalogTransformerException {
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
```

Comment:
Make this a try-with-resources to ensure the BAOS is closed when you're done with it.